internal/apischema/openai/openai.go

```diff
@@ -276,6 +321,14 @@ func (c *ChatCompletionMessageParamUnion) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// This function marshals the ChatCompletionMessageParamUnion into JSON.
+func (c *ChatCompletionMessageParamUnion) MarshalJSON() ([]byte, error) {
+	if c.Value == nil {
+		return nil, fmt.Errorf("cannot marshal ChatCompletionMessageParamUnion with nil value")
+	}
+	return json.Marshal(c.Value)
```

Comment:
yep, at least the repro i shared ^^ will be resolved with 
```suggestion
	switch c.Type {
	case ChatMessageRoleUser:
		if userMessage, ok := c.Value.(ChatCompletionUserMessageParam); ok {
			return json.Marshal(&userMessage)
		}
	case ChatMessageRoleAssistant:
		if assistantMessage, ok := c.Value.(ChatCompletionAssistantMessageParam); ok {
			return json.Marshal(&assistantMessage)
		}
	case ChatMessageRoleSystem:
		if systemMessage, ok := c.Value.(ChatCompletionSystemMessageParam); ok {
			return json.Marshal(&systemMessage)
		}
	case ChatMessageRoleDeveloper:
		if developerMessage, ok := c.Value.(ChatCompletionDeveloperMessageParam); ok {
			return json.Marshal(&developerMessage)
		}
	case ChatMessageRoleTool:
		if toolMessage, ok := c.Value.(ChatCompletionToolMessageParam); ok {
			return json.Marshal(&toolMessage)
		}
	}
	return nil, fmt.Errorf("cannot marshal ChatCompletionMessageParamUnion with unknown type: %s", c.Type)
```